### PR TITLE
Blobs support typed arrays now

### LINKED ISSUE #20348: Implement support for blobs created from typed arrays
Now that #20267 has merged we can implement APIs that accept typed arrays as arguments. The ArrayBuffer and ArrayBufferView types are commented out in the BlobPart union, so uncommenting them means we need to modify the blob constructor to deal with typed arrays.

Code: `components/script/dom/webidls/Blob.webidl`, `components/script/dom/blob.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/FileAPI` (there should be some new [passing tests](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations))

### PR DESCRIPTION
**Pull request by pvdrz**
Blobs support typed arrays now, the relevant test were modified

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #20348 (github issue number if applicable).

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20370)

### CODE REVIEW

File: tests/wpt/metadata/FileAPI/blob/Blob-slice.html.ini [hunk 1]
```diff
@@ -1,62 +1,62 @@
 [Blob-slice.html]
   type: testharness
   [Slicing test: slice (5,0).]
-    expected: FAIL
+    expected: PASS
```
Comment (CYBAI) [inline tests/wpt/metadata/FileAPI/blob/Blob-slice.html.ini]: When we'd like to make the test as `PASS`, we can just remove the `expected: FAIL` to make it as `PASS`

Ex.
```diff
[Blob-slice.html]
   type: testharness
-   [Slicing test: slice (5,0).]
-     expected: FAIL
-     bug: https://github.com/servo/servo/issues/10911
```

Comment (christianpoveda) [reply tests/wpt/metadata/FileAPI/blob/Blob-slice.html.ini]: Ok let me fix it

File: tests/wpt/metadata/FileAPI/blob/Blob-constructor.html.ini [hunk 2]
```diff
@@ -5,32 +5,20 @@
     bug: https://github.com/servo/rust-mozjs/issues/269
 
   [ArrayBuffer elements of the blobParts array should be supported.]
```
Comment (jdm) [inline tests/wpt/metadata/FileAPI/blob/Blob-constructor.html.ini]: The names of the subtests that now pass should be removed too.

**Review by jdm — CHANGES_REQUESTED** [review]
This looks good apart from this comment!

**Comment by jdm** [comment]
Please [squash](https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/) all the commits together, too.

File: tests/wpt/metadata/FileAPI/blob/Blob-constructor.html.ini (hunk 2 above)
Comment (christianpoveda) [reply tests/wpt/metadata/FileAPI/blob/Blob-constructor.html.ini]: What happens if a test ends up without subtests in its .ini file? should I leave the file?

Comment (jdm) [reply tests/wpt/metadata/FileAPI/blob/Blob-constructor.html.ini]: Nope! The file can be removed.

**Comment by jdm** [comment]
@bors-servo r+
Thanks for fixing this!

**Pull request merged**